src/java/src/main/java/com/dnanexus/DXDataObject.java

```diff
@@ -1062,7 +1062,7 @@ public Describe getCachedDescribe() {
      * @return a DNAnexus link
      */
     @JsonValue
-    private JsonNode getDXLink() {
+    public JsonNode getDXLink() {
```

Comment:
Let's call this `getLinkAsJson` since we might want to have a `getDXLink` in the future that returns some native class (implemented by us) representing a link.

Take the part about the JSON serializer and remove it from the public docstring for this method. You could move it in some form to the main docstring for this class...

> Instances of this class are JSON-serialized as a DNAnexus link, which makes it suitable to provide an objects with references to DXDataObjects as the input when running a DXApplet, etc.